File: backend/app/schemas/order.py
from pydantic import BaseModel
from typing import Optional, List
from uuid import UUID
from datetime import datetime
from app.db.models import OrderStatus

class OrderItemBase(BaseModel):
    product_id: UUID
    quantity: int
    unit_price: float

class OrderItemCreate(OrderItemBase):
    pass

class OrderItem(OrderItemBase):
    id: UUID
    order_id: UUID
    total_price: float

    class Config:
        from_attributes = True

class OrderBase(BaseModel):
    cafe_id: UUID
    notes: Optional[str] = None

class OrderCreate(OrderBase):
    items: List[OrderItemCreate]

class OrderUpdate(BaseModel):
    status: Optional[OrderStatus] = None
    notes: Optional[str] = None

class OrderInDBBase(OrderBase):
    id: UUID
    user_id: UUID
    status: OrderStatus
    total_amount: float
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class Order(OrderInDBBase):
    items: List[OrderItem] = []

class OrderInDB(OrderInDBBase):
    pass

File: frontend/src/types/order.ts
export interface OrderItem {
  id: string;
  order_id: string;
  product_id: string;
  quantity: number;
  unit_price: number;
  total_price: number;
}

export interface OrderItemCreate {
  product_id: string;
  quantity: number;
  unit_price: number;
}

export interface Order {
  id: string;
  user_id: string;
  business_id: string;
  status: OrderStatus;
  total_amount: number;
  notes?: string;
  created_at: string;
  updated_at?: string;
  items: OrderItem[];
}

export interface OrderCreate {
  business_id: string;
  notes?: string;
  items: OrderItemCreate[];
}

export interface OrderUpdate {
  status?: string;
  notes?: string;
}

export enum OrderStatus {
  PENDING = "pending",
  CONFIRMED = "confirmed",
  PREPARING = "preparing",
  READY = "ready",
  DELIVERED = "delivered",
  CANCELLED = "cancelled"
}

export interface CartItem {
  product: {
    id: string;
    name: string;
    price: number;
    business_id: string;
    description?: string;
    image_url?: string;
  };
  quantity: number;
}